compiler/x/codegen/UnaryEvaluator.cpp

```diff
@@ -36,15 +36,74 @@
 
 extern TR::Register *intOrLongClobberEvaluate(TR::Node *node, bool nodeIs64Bit, TR::CodeGenerator *cg);
 
+TR::Register *OMR::X86::TreeEvaluator::floatingPointAbsHelper(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *valueNode = node->getChild(0);
+   TR::Register *resultReg = cg->allocateRegister(TR_VRF);
+   TR::DataType et = node->getType().getVectorElementType();
+   TR::VectorLength vl = node->getType().getVectorLength();
+   TR::Register *valueReg = cg->evaluate(valueNode);
+
+   TR::InstOpCode andOpcode = TR::InstOpCode::PANDRegReg;
+   TR::InstOpCode shrOpcode = et == TR::Double ? TR::InstOpCode::PSRLQRegImm1 : TR::InstOpCode::PSRLDRegImm1;
+   OMR::X86::Encoding shrEncoding = shrOpcode.getSIMDEncoding(&cg->comp()->target().cpu, vl);
+   OMR::X86::Encoding andEncoding = andOpcode.getSIMDEncoding(&cg->comp()->target().cpu, vl);
+
+   TR_ASSERT_FATAL(shrEncoding != OMR::X86::Bad, "vabs: No encoding method for shift opcode");
+   TR_ASSERT_FATAL(andEncoding != OMR::X86::Bad, "vabs: No encoding method for and opcode");
+
+   if (vl == TR::VectorLength512)
```

Comment:
Being a newbie to the SIMD instructions, it took me a fair bit of time to figure out how the instruction sequences that are being generated implement VABS.  I think a short, high-level description of how this works might help the next person who passes through here.